src/env.rs

```diff
@@ -7,6 +10,19 @@ use crate::error::Result;
 use crate::map::Map;
 use crate::source::Source;
 use crate::value::{Value, ValueKind};
+use crate::ConfigError;
+
+/// Failed to decode an environment variable value as a Unicode string.
+#[derive(Debug)]
+pub struct InvalidUnicode(OsString);
```

Comment:
I first tried to find a stdlib error to return, but the APIs returning the conversion error of OsString -> String seem to be OS-specific, so I added this generic one. (that's also why I report it via `ConfigError::Foreign`, perhaps another variant is preferred? and I could move this to `error.rs`)